LocationProvider "gps" not found

### App information
* ID: de.dennisguse.opentracks
* Version: 5017 v4.0.8

### Device information
* Brand: OnePlus
* Device: OnePlus5T
* Model: ONEPLUS A5010
* Id: TP1A.220624.021.A1
* Product: OnePlus5T

### Firmware
* SDK: 33
* Release: 13
* Incremental: 1661937071

### Cause of error
```java
Exception in thread "main": java.lang.IllegalArgumentException: provider "gps" does not exist
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3015)
	at android.os.Parcel.createException(Parcel.java:2995)
	at android.os.Parcel.readException(Parcel.java:2978)
	at android.os.Parcel.readException(Parcel.java:2920)
	at android.location.ILocationManager$Stub$Proxy.registerLocationListener(ILocationManager.java:1291)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1551)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1234)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1199)
	at android.location.LocationManager.requestLocationUpdates(LocationManager.java:1170)
	at de.dennisguse.opentracks.services.handlers.GPSHandler.registerLocationListener(GPSHandler.java:157)
	at de.dennisguse.opentracks.services.handlers.GPSHandler.onStart(GPSHandler.java:45)
	at de.dennisguse.opentracks.services.handlers.TrackPointCreator.start(TrackPointCreator.java:55)
	at de.dennisguse.opentracks.services.TrackRecordingService.startSensors(TrackRecordingService.java:208)
	at de.dennisguse.opentracks.services.TrackRecordingService.startRecording(TrackRecordingService.java:195)
	at de.dennisguse.opentracks.services.TrackRecordingService.startNewTrack(TrackRecordingService.java:175)
	at de.dennisguse.opentracks.TrackListActivity.lambda$onCreate$4$de-dennisguse-opentracks-TrackListActivity(TrackListActivity.java:237)
	at de.dennisguse.opentracks.TrackListActivity$$ExternalSyntheticLambda0.onConnected(Unknown Source:2)
	at de.dennisguse.opentracks.services.TrackRecordingServiceConnection.setTrackRecordingService(TrackRecordingServiceConnection.java:147)
	at de.dennisguse.opentracks.services.TrackRecordingServiceConnection.onServiceConnected(TrackRecordingServiceConnection.java:162)
	at android.app.LoadedApk$ServiceDispatcher.doConnected(LoadedApk.java:2188)
	at android.app.LoadedApk$ServiceDispatcher$RunConnection.run(LoadedApk.java:2221)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7898)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.internal.util.Preconditions.checkArgument(Preconditions.java:59)
	at com.android.server.location.LocationManagerService.registerLocationListener(LocationManagerService.java:722)
	at android.location.ILocationManager$Stub.onTransact(ILocationManager.java:582)
	at android.os.Binder.execTransactInternal(Binder.java:1280)
	at android.os.Binder.execTransact(Binder.java:1244)

```

--- a/src/main/java/de/dennisguse/opentracks/services/handlers/GPSHandler.java
+++ b/src/main/java/de/dennisguse/opentracks/services/handlers/GPSHandler.java
@@ -10,6 +10,7 @@ import android.util.Log;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.VisibleForTesting;
+import androidx.core.location.LocationManagerCompat;
 
 import java.time.Duration;
 import java.time.Instant;
@@ -152,6 +153,12 @@ public class GPSHandler implements LocationListener, GpsStatus.GpsStatusListener
             Log.e(TAG, "Not started.");
             return;
         }
+
+        if(!LocationManagerCompat.hasProvider(locationManager, LocationManager.GPS_PROVIDER)) {
+            Log.e(TAG, "Device doesn't have GPS.");
+            return;
+        }
+
         if (PermissionRequester.GPS.hasPermission(context)) {
             try {
                 locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, gpsInterval.toMillis(), 0, this);

